app/public/src/pages/preparation.tsx

```diff
@@ -201,6 +204,18 @@ export default function Preparation() {
           dispatch(leavePreparation())
           navigate("/lobby")
           playSound(SOUNDS.LEAVE_ROOM)
+        } else if (shouldReconnect) {
+          logger.log("Connection closed unexpectedly or timed out. Attempting reconnect.")
+          // Restart the expiry timer of the reconnection token and clear room data to refresh
+          localStore.set(
+            LocalStoreKeys.RECONNECTION_PREPARATION,
+            { reconnectionToken: r.reconnectionToken, roomId: r.roomId },
+            30
+          )
+          initialized.current = false
+          dispatch(leavePreparation())
+          // Update dependency variable to force a re-render
+          setReconnectTries(x => x + 1)
```

Comment:
same comment than for game, you don't need to rerender. It's much easier for preparation, you can just call the `reconnect()` function that takes care of everything. We could do a similar `reconnect` function for game room as well